sdk/iOS/OoyalaSkinSDK/React/OOReactSkinModel.m

```diff
@@ -0,0 +1,352 @@
+}
+
+- (void)dealloc {
+  [[NSNotificationCenter defaultCenter] removeObserver:self];
+  [OOVolumeManager removeVolumeObserver:self];
+}
+
+#pragma mark - RCTBridgeDelegate
+
+- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge {
+  return self.skinOptions.jsCodeLocation;
+}
+
+#pragma mark - OOReactSkinBridgeDelegate
+// Binding ooReactSkinModel as a delegate for handling RN events
+- (void)bridge:(OOReactSkinBridge *)bridge didLoadModule:(id<OOReactSkinBridgeModule>)module {
+  if ([module isKindOfClass:[OOReactSkinBridgeModuleMain class]]) {
+    [(OOReactSkinBridgeModuleMain*)module setSkinViewDelegate:self];
+  }
+}
+
+#pragma mark - OOReactSkinModelDelegate
+
+- (void)handleAudioTrackSelection:(NSString *)audioTrackName {
+  // TODO: Can we move this logic to core?
+  for (id<OOAudioTrackProtocol> audioTrack in [self.player availableAudioTracks]) {
+    // This check is a temporary solution
+    // Need to wait conclusion about tracks names (how need create them) from Ooyala
+    if ([audioTrack.title isEqualToString:audioTrackName]) {
+      [self.player setDefaultAudioTrack:audioTrack];
+      [self.player setAudioTrack:audioTrack];
+      return;
+    }
+  }
+
+  LOG(@"handleAudioTrackSelection - Can't find audio track");
+}
+
+- (void)handlePlaybackSpeedRateSelection:(nullable NSNumber *)selectedPlaybackSpeedRate {
+  if (selectedPlaybackSpeedRate) {
+    [self.player changePlaybackSpeedRate:selectedPlaybackSpeedRate.floatValue];
+  } else {
+    LOG(@"handlePlaybackSpeedRateSelection - selectedPlaybackSpeedRate invalid type");
+  }
+}
+
+- (void)handleDiscoveryClick:(NSString *)bucketInfo embedCode:(NSString *)embedCode {
+  [OODiscoveryManager sendClick:self.skinOptions.discoveryOptions bucketInfo:bucketInfo pcode:self.player.pcode parameters:nil];
+  [self.player setEmbedCode:embedCode];
+  [self.player play];
+}
+
+- (void)handleDiscoveryImpress:(NSString *)bucketInfo {
+  [OODiscoveryManager sendImpression:self.skinOptions.discoveryOptions bucketInfo:bucketInfo pcode:self.player.pcode parameters:nil];
+}
+
+- (void)handleIconClick:(NSInteger)index {
+  [self.player onAdIconClicked:index];
+}
+
+- (void)handleLanguageSelection:(NSString *)language {
+  [self.player setClosedCaptionsLanguage:language];
+}
+
+- (void)handleLearnMore {
+  [self.player clickAd];
+}
+
+- (void)handleMoreOption {
+  [self.player pause];
+}
+
+- (void)handleOverlay:(NSString *)url {
+  [self.player onAdOverlayClicked:url];
+}
+
+- (void)handlePip {
+  [self.player togglePictureInPictureMode];
+}
+
+- (void)handlePlay {
+  [self.player play];
+}
+
+- (void)handlePlayPause {
+  self.player.state == OOOoyalaPlayerStatePlaying ? [self.player pause] : [self.player play];
+}
+
+- (void)playPauseFromAdTappedNotification {
+  if (![self.skinControllerDelegate isReactViewInteractionEnabled]) {
+    [self handlePlayPause];
+  }
+}
+
+- (void)handleReplay {
+  [self.player seek:0];
+  [self.player play];
```

Comment:
Do we need to do anything else here so that ads replay when you watch the video again?

Reply:
@cacel could you please answer?